api/v1beta1/disk_failure.go

```diff
@@ -25,28 +28,64 @@ type DiskFailureSpec struct {
 	Paths []string `json:"paths"`
 	// +nullable
 	OpenatSyscall *OpenatSyscallSpec `json:"openat,omitempty"`
+	// +kubebuilder:validation:Required
+	// +ddmark:validation:Required=true
+	Probability string `json:"probability"`
 }
 
 // MaxDiskPathCharacters is used to limit the number of characters due to the eBPF memory kernel limitation.
 const MaxDiskPathCharacters = 62
 
 // Validate validates args for the given disruption
-func (s *DiskFailureSpec) Validate() error {
+func (s *DiskFailureSpec) Validate() (retErr error) {
+	if err := s.validatePaths(); err != nil {
+		retErr = multierror.Append(retErr, err)
+	}
+
+	if err := s.validateProbability(); err != nil {
+		retErr = multierror.Append(retErr, err)
+	}
+
+	return
+}
+
+func (s *DiskFailureSpec) validatePaths() error {
 	for _, path := range s.Paths {
-		path := strings.TrimSpace(path)
+		trimSpacePath := strings.TrimSpace(path)
 
-		if path == "" {
+		if trimSpacePath == "" {
 			return fmt.Errorf("the path of the disk failure disruption must not be empty")
 		}
 
-		if len(path) > MaxDiskPathCharacters {
+		if len(trimSpacePath) > MaxDiskPathCharacters {
 			return fmt.Errorf("the path of the disk failure disruption must not exceed %d characters, found %d", MaxDiskPathCharacters, len(path))
 		}
 	}
 
 	return nil
 }
 
+func (s *DiskFailureSpec) validateProbability() error {
+	probabilityError := fmt.Errorf("the probability of the disk failure disruption should be a percentage within the range of 1%% to 100%%")
+
+	if !strings.HasSuffix(s.Probability, "%") {
+		return probabilityError
```

Comment:
As we are testing explicitly for all the error paths, should we have unique error instead to be explicit about why we are rejecting the disruption?
```suggestion
		return fmt.Errorf("the probability of the disk failure disruption should be a percentage with a %% suffix")
```

Reply:
💯 agree with you!  I prefer to be explicit rather implicit 👍